src/App.tsx

```diff
@@ -17,18 +69,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onQueryChange={onQueryChange}
+                onReset={onReset}
+                onStatusChange={onStatusChange}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loadingTodos && todos.length === 0 && <Loader />}
+              {!loadingTodos && (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodoId}
+                  onOpenModal={onOpenModal}
+                />
+              )}
```

Comment:
Will be more readable and better

```suggestion
              {(loadingTodos && !todos.length) ? (
                <Loader />
              ) : (
                <TodoList
                  todos={filteredTodos}
                  selectedTodoId={selectedTodoId}
                  onOpenModal={onOpenModal}
                />
              )}
```